src/MizarWidgets/include/MizarWidgets/SamplingWithFrameTrackReportModel.h

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef MIZAR_WIDGETS_SAMPLING_WITH_FRAME_TRACK_REPORT_MODEL_H_
+#define MIZAR_WIDGETS_SAMPLING_WITH_FRAME_TRACK_REPORT_MODEL_H_
+
+#include <absl/container/flat_hash_map.h>
+
+#include <QAbstractItemModel>
+#include <QVariant>
+#include <Qt>
+#include <algorithm>
+#include <iterator>
+#include <vector>
+
+#include "MizarData/SamplingWithFrameTrackComparisonReport.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/Typedef.h"
+
+namespace orbit_mizar_widgets {
+
+template <typename Report, typename Counts, typename FrameTrackStats>
+class SamplingWithFrameTrackReportModelTmpl : public QAbstractTableModel {
+  template <typename T>
+  using Baseline = ::orbit_mizar_base::Baseline<T>;
+  template <typename T>
+  using Comparison = ::orbit_mizar_base::Comparison<T>;
+  using SFID = ::orbit_mizar_base::SFID;
+
+ public:
+  enum class Column {
+    kFunctionName,
+    kBaselineExclusivePercent,
+    kBaselineExclusiveTimePerFrame,
+    kComparisonExclusivePercent,
+    kComparisonExclusiveTimePerFrame,
+    kPvalue,
+    kIsSignificant,
+    kSlowdownPercent,
+    kSlowdownPerFrame
+  };
+
+  static constexpr int kColumnsCount = 9;
+
+  explicit SamplingWithFrameTrackReportModelTmpl(Report report, QObject* parent = nullptr)
+      : QAbstractTableModel(parent), report_(std::move(report)) {
+    for (const auto& [sfid, unused_name] : report_.GetSfidToNames()) {
+      if (*BaselineExclusiveCount(sfid) > 0 || *ComparisonExclusiveCount(sfid) > 0) {
+        sfids_.push_back(sfid);
+      }
+    }
+  }
+
+  [[nodiscard]] int rowCount(const QModelIndex& /*parent*/) const override {
+    return static_cast<int>(sfids_.size());
+  };
+  [[nodiscard]] int columnCount(const QModelIndex& /*parent*/) const override {
+    return kColumnsCount;
+  };
+  [[nodiscard]] QVariant data(const QModelIndex& index, int role) const override {
+    if (role != Qt::DisplayRole) return {};
+    return QString::fromStdString(MakeDisplayedString(index));
+  }
+
+  [[nodiscard]] QVariant headerData(int section, Qt::Orientation orientation,
+                                    int role) const override {
```

Comment:
All these model functions have default arguments in Qt. (Like `QModelIndex{}` for the index in `rowCount`.)

We don't want to have default arguments for virtual functions because they can get out of sync - like here. Could you sync them up since we can't change them in Qt?